File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/custome/Custom4D2FRemoteControl.java
package com.example.try_gameengine.remotecontroller.custome;

import android.view.MotionEvent;

public class Custom4D2FRemoteControl {
	Custom4D2FCommand slot;
	
	Custom4D2FCommand[] onCommands;
	Custom4D2FCommand[] offCommands;

	public Custom4D2FRemoteControl() {
		onCommands = new Custom4D2FCommand[7];
		offCommands = new Custom4D2FCommand[7];
		Custom4D2FCommand noCommand = new Custom4D2FNoCommand();
		for(int i=0; i<7; i++){
			onCommands[i] = noCommand;
			offCommands[i] = noCommand;
		}
	}
	
	public void setCommand(Custom4D2FCommand command){
		slot = command;
	}
	
	public void setCommand(int slot, Custom4D2FCommand onCommand, Custom4D2FCommand offCommand){
		onCommands[slot] = onCommand;
		offCommands[slot] = offCommand;
	}
	
	public Custom4D2FCommandType executePressDown(float x, float y, int motionEventPointerId, MotionEvent event){
		Custom4D2FCommandType commandType = Custom4D2FCommandType.None;
		for(Custom4D2FCommand command : onCommands){
			if(command.checkExecute(x, y, event)){
				commandType = command.execute();
				command.setMotionEventPointerId(motionEventPointerId);
				break;
			}
		}
		return commandType;
	}
	
	public Custom4D2FCommandType executePressUp(float x, float y, int motionEventPointerId, MotionEvent event){
		Custom4D2FCommandType commandType = Custom4D2FCommandType.None;
		
		for(Custom4D2FCommand command : offCommands){
			if(command.checkExecute(x, y, event)){
				commandType = command.execute();
				break;
			}
		}

		return commandType;
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/custome/Custom4D2FRemoteControl.kt
package com.example.try_gameengine.remotecontroller.custome

import android.view.MotionEvent

class Custom4D2FRemoteControl {
    var slot: Custom4D2FCommand? = null

    var onCommands: Array<Custom4D2FCommand>
    var offCommands: Array<Custom4D2FCommand>

    init {
        val noCommand: Custom4D2FCommand = Custom4D2FNoCommand()
        onCommands = Array(7) { noCommand }
        offCommands = Array(7) { noCommand }
    }

    fun setCommand(command: Custom4D2FCommand?) {
        slot = command
    }

    fun setCommand(slot: Int, onCommand: Custom4D2FCommand?, offCommand: Custom4D2FCommand?) {
        onCommands[slot] = onCommand!!
        offCommands[slot] = offCommand!!
    }

    fun executePressDown(
        x: Float,
        y: Float,
        motionEventPointerId: Int,
        event: MotionEvent?
    ): Custom4D2FCommandType? {
        var commandType: Custom4D2FCommandType? = Custom4D2FCommandType.None
        for (command in onCommands) {
            if (command.checkExecute(x, y, event)) {
                commandType = command.execute()
                command.setMotionEventPointerId(motionEventPointerId)
                break
            }
        }
        return commandType
    }

    fun executePressUp(
        x: Float,
        y: Float,
        motionEventPointerId: Int,
        event: MotionEvent?
    ): Custom4D2FCommandType? {
        var commandType: Custom4D2FCommandType? = Custom4D2FCommandType.None

        for (command in offCommands) {
            if (command.checkExecute(x, y, event)) {
                commandType = command.execute()
                break
            }
        }

        return commandType
    }
}
